kano_updater/utils.py

```diff
@@ -677,3 +679,58 @@ def disable_power_button():
     except Exception:
         # Kano Peripherals doesn't support this function
         pass
+
+
+def verify_kit_is_plugged():
+    """
+    On Computer Kit 2 Pro, verify that the battery is plugged in and not depleting.
+    For now, we rely on the user to tell us.
+
+    Returns:
+        is_plugged - bool whether the kit is plugged in or not.
+    """
+    ck2_pro = False
+    is_battery_low = False
+    is_plugged = True
+
+    try:
+        from kano_peripherals.wrappers.detection import is_ck2_pro
+        ck2_pro = is_ck2_pro()
+
+        from kano_peripherals.ck2_pro_hat.driver.high_level import get_ck2_pro_hat_interface
+        ck2pro_iface = get_ck2_pro_hat_interface()
+        is_battery_low = (ck2pro_iface and ck2pro_iface.is_battery_low())
+    except:
+        # Kano Peripherals doesn't support this function
+        pass
+
+    if ck2_pro:
+        # Run the first dialog asking the user a question.
+        dialog = KanoDialog(
+            title_text=_("Power Required"),
+            description_text=_("Is your computer plugged in?"),
+            button_dict={
+                _("Yes"): {'color': 'green', 'return_value': True},
+                _("No"): {'color': 'red', 'return_value': False}
+            }
+        )
+        is_plugged = dialog.run()
+
+        header = ""
+        if is_battery_low:
```

Comment:
After the previous message, it is likely that the user will go and find their cable and plug it in, in which case `is_battery_low` will have changed since then so we might actually want to do the check here.